src/Microsoft.OData.Core/Uri/NodeToStringBuilder.cs

```diff
@@ -470,7 +471,245 @@ internal String TranslateNode(QueryNode node)
+            string withinBrackets = subtrings[1];
+            char[] parameterSeparators = { ',' };
+            string[] parameters = withinBrackets.Trim().Split(parameterSeparators, StringSplitOptions.RemoveEmptyEntries);
+            string leftParameter = parameters[0];
+            string rightParameter = parameters.Length == 2 ? parameters[1] : String.Empty;
+
+            if (firstParameterRangeVariableNode != null && IsDifferentSource(filterClauseRangeVariable, firstParameterRangeVariableNode))
+            {
+                leftParameter = ExpressionConstants.It + ExpressionConstants.SymbolForwardSlash + leftParameter;
+                translatedNode = parameters.Length == 1 ? subtrings[0] + '(' + leftParameter + ')' : subtrings[0] + '(' + leftParameter + ',' + rightParameter + ')';
+            }
+
+            return translatedNode;
+        }
+
+        /// <summary>
+        /// Bind $it to the <see cref="SingleValueOpenPropertyAccessNode"/> translated string.
+        /// </summary>
+        /// <param name="node">node to bind.</param>
+        /// <param name="filterClauseRangeVariable">The <see cref="FilterClause"/> range variable.</param>
+        /// <returns>The translated string with $it binding.</returns>
+        private string BindSingleValueOpenPropertyAccess(SingleValueOpenPropertyAccessNode node, ResourceRangeVariable filterClauseRangeVariable)
+        {
+            string translatedNode = this.TranslateNode(node);
+            ResourceRangeVariableReferenceNode rangeVariableNode = GetResourceRangeVariableReferenceNode(node);
+
+            if (rangeVariableNode != null && IsDifferentSource(filterClauseRangeVariable, rangeVariableNode))
+            {
+                translatedNode = ExpressionConstants.It + ExpressionConstants.SymbolForwardSlash + translatedNode;
+            }
+
+            return translatedNode;
+        }
+
+        /// <summary>
+        /// Check whether Navigation source of the FilterClause rangeVariable is different from the Expression rangeVariable.
+        /// </summary>
+        /// <param name="filterClauseRangeVariable"><see cref="FilterClause"/> rangeVariable.</param>
+        /// <param name="rangeVariableReferenceNode">The rangeVariable of the expression.</param>
+        /// <returns>If Navigation Source are different, returns true. Otherwise false.</returns>
+        private static bool IsDifferentSource(ResourceRangeVariable filterClauseRangeVariable, ResourceRangeVariableReferenceNode rangeVariableReferenceNode)
+        {
+            Debug.Assert(filterClauseRangeVariable != null, "filterClause != null");
+            Debug.Assert(rangeVariableReferenceNode != null, "rangeVariableReferenceNode != null");
+
+            if (filterClauseRangeVariable.NavigationSource != rangeVariableReferenceNode.NavigationSource)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }

```

Comment:
could simplify to:
   return filterClauseRangeVariable.NavigationSource != rangeVariableReferenceNode.NavigationSource;